packages/entities/src/lib/update.mutation.ts

```diff
@@ -243,3 +243,85 @@ export function upsertEntities<
     };
   };
 }
+
+/**
+ * Update entities ids
+ *
+ * @example
+ *
+ * // update single entity id
+ * store.update(updateEntitiesIds(1, 2));
+ *
+ * // update multiple entities ids
+ * store.update(updateEntitiesIds([1, 2], [10, 20]));
+ *
+ * // update entity id using a custom ref
+ * store.update(updateEntitiesIds(1, 2, { ref: UIEntitiesRef }));
+ *
+ */
+export function updateEntitiesIds<
+  S extends EntitiesState<Ref>,
+  Ref extends EntitiesRef = DefaultEntitiesRef
+>(
+  oldId: OrArray<getIdType<S, Ref>>,
+  newId: OrArray<getIdType<S, Ref>>,
+  options: BaseEntityOptions<Ref> = {}
+): Reducer<S> {
+  return function (state, context) {
+    const oldIds = coerceArray(oldId);
+    const newIds = coerceArray(newId);
+
+    if (oldIds.length !== newIds.length) {
+      throw new Error('The number of old and new ids must be equal');
+    }
+
+    const { ref = defaultEntitiesRef } = options;
+    const idProp = getIdKey<string>(context, ref);
+    const updatedEntities = { ...state[ref.entitiesKey] };
+
+    for (let i = 0; i < oldIds.length; i++) {
+      const oldVal = oldIds[i];
+      const newVal = newIds[i];
+
+      if (state[ref.entitiesKey][newVal]) {
+        throw new Error(
+          `Updating id "${oldVal}". The new id "${newVal}" already exists`
+        );
+      }
+
+      const oldEntity = state[ref.entitiesKey][oldVal];
+      const updated = { ...oldEntity, [idProp]: newVal };
+
+      updatedEntities[newVal] = updated;
+      Reflect.deleteProperty(updatedEntities, oldVal);
+    }
+
+    const updatedStateIds: getIdType<S, Ref>[] = [...state[ref.idsKey]];
```

Comment:
```suggestion
    const currentIds: getIdType<S, Ref>[] = [...state[ref.idsKey]];
```

Reply:
except that these are not current ids. We updated the reference to the array and we'll be changing some of the ids in that array too. Later on, when we return updated state, the `currentIds` looks weird:
```ts
return {
  ...state,
  [ref.entitiesKey]: updatedEntities,
  [ref.idsKey]: currentIds,
};
```